staging/src/k8s.io/kubectl/pkg/cmd/explain/explain.go

```diff
@@ -130,10 +131,43 @@ func (o *ExplainOptions) Run(args []string) error {
 
 	gvk, _ := o.Mapper.KindFor(fullySpecifiedGVR)
```

Comment:
The problem is that this will always return a GroupVersionKind, all of our API is prioritized, so with multiple APIs like from your example the one with higher priority will match. So you won't reach the code inside at all. I think you'd need to move the logic outside the below condition.

Reply:
It does not matter as long as all the APIs are returned. Let's assume the list of returned APIs is (in the given order, some APIs are fabricated):
- authentications.config.openshift.io (K=Authentications, G=config.openshift.io)
- authentications.operator.openshift.io (K=Authentications, G=operator.openshift.io)
- authentications.migration.k8s.io (K=Authentications, G=migration.k8s.io)
- authentications.metal3.io (K=Authentications, G=metal3.io)
- authentications.whereabouts.cni.cncf.io (K=Authentications, G=whereabouts.cni.cncf.io)
- authentications.snapshot.storage.k8s.io (K=Authentications, G=snapshot.storage.k8s.io)

> with multiple APIs like from your example the one with higher priority will match. 

Right, so you need to specify the group as well (e.g. `kubectl explain authentications.metal3.io` ). In which case the suggested changes will go through all APIs and pick the one which corresponds to the group.

> So you won't reach the code inside at all.

Right, `kubectl explain authentications.metal3.io.field` will not find anything since there's no `metal3.io.field` group. In which case you will need to specify `--api-version=metal3.io/VERSION` flag which will bypass the new code.

Can you reach any code (=field) inside with the current implementation? No.

Reply:
For having all GVK, you'd need to use `KindsFor` rather than `KindFor`, notice `s` after kind in the first method. That will return you all the known GVKs. What I was saying before, though, is that few lines earlier in `SplitAndParseResourceRequest` there's already parsing and matching with mapper which might result in only a single value. That's why I was proposing to move the code a bit earlier.